core/shared/platform/windows/win_file.c

```diff
@@ -275,17 +352,17 @@ create_handle(wchar_t *path, bool is_dir, bool follow_symlink, bool readonly)
 
     DWORD desired_access = GENERIC_READ;
 
-    if (!readonly) {
+    if (!readonly)
         desired_access |= GENERIC_WRITE;
+    else
         create_params.dwFileAttributes |= FILE_ATTRIBUTE_READONLY;
-    }
 
     return CreateFile2(path, desired_access,
                        FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
                        OPEN_EXISTING, &create_params);
 }
 
-#if WINAPI_PARTITION_DESKTOP
+#if !WINAPI_PARTITION_DESKTOP
```

Comment:
Suggest changing to `WINAPI_PARTITION_DESKTOP == 0` to make code easier to understand.